File: Backend/Travle.Model/Responses/PaymentIntentResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// What the client needs to present the Stripe PaymentSheet for a booking. <see cref="ClientSecret"/>
    /// is the per-intent secret the mobile SDK confirms against; <see cref="PublishableKey"/> is the
    /// non-secret <c>pk_test_…</c> (the app also has it via <c>--dart-define</c>, echoed here for
    /// convenience). The amounts are the server-computed source of truth, shown for confirmation only —
    /// success is recorded solely by the signature-verified webhook, never by the client.
    /// </summary>
    public class PaymentIntentResponse
    {
        public int BookingId { get; set; }

        /// <summary>Our Payment row id (the Stripe intent id is never exposed to the client).</summary>
        public int PaymentId { get; set; }

        public string ClientSecret { get; set; } = string.Empty;
        public string PublishableKey { get; set; } = string.Empty;

        /// <summary>Charged amount in major units (KM), for display; e.g. 25.00.</summary>
        public decimal Amount { get; set; }
        public string Currency { get; set; } = "bam";
    }
}

File: UI/travle_core/lib/src/models/payment_intent_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_intent_response.g.dart';

/// What the client needs to present the Stripe PaymentSheet for a booking
/// (mirrors the backend `PaymentIntentResponse`). [clientSecret] is the
/// per-intent secret the Stripe SDK confirms against; [publishableKey] is the
/// non-secret `pk_test_…` the SDK is initialised with. [amount]/[currency] are
/// the server-computed source of truth, shown for confirmation only — payment
/// success is recorded solely by the signature-verified webhook, never here.
@JsonSerializable()
class PaymentIntentResponse {
  PaymentIntentResponse({
    required this.bookingId,
    required this.paymentId,
    required this.clientSecret,
    required this.publishableKey,
    required this.amount,
    required this.currency,
  });

  final int bookingId;
  final int paymentId;
  final String clientSecret;
  final String publishableKey;
  final double amount;
  final String currency;

  factory PaymentIntentResponse.fromJson(Map<String, dynamic> json) =>
      _$PaymentIntentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentIntentResponseToJson(this);
}
